### PR TITLE
Consolidate example to_date.rs into dateframe.rs

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

- Part of https://github.com/apache/datafusion/issues/11172

## Rationale for this change

1. Make it easier to find relevant examples
2. Make local dev faster with fewer distinct binary targets (each example binary takes 100+ MB to compile)

## What changes are included in this PR?
1. Consolidate to_date.rs into dataframe.rs

## Are these changes tested?
Run by CI

### CODE REVIEW

File: datafusion-examples/examples/dataframe.rs [hunk 1]
```diff
@@ -28,16 +28,24 @@ use std::io::Write;
 use std::sync::Arc;
 use tempfile::tempdir;
 
-/// This example demonstrates using DataFusion's DataFrame API to
+/// This example demonstrates using DataFusion's DataFrame API
```
Comment (alamb) [inline datafusion-examples/examples/dataframe.rs:31]: Note the `to_date` function isn't even listed in the readme menu: https://github.com/apache/datafusion/blob/main/datafusion-examples/README.md

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

File: datafusion-examples/examples/dataframe.rs [hunk 2]
```diff
@@ -206,3 +215,38 @@ async fn write_out(ctx: &SessionContext) -> std::result::Result<(), DataFusionEr
...
+    ctx.register_batch("t", batch)?;
+    let df = ctx.table("t").await?;
+
+    // use to_date function to convert col 'a' to timestamp type using the default parsing
+    let df = df.with_column("a", to_date(vec![col("a")]))?;
+
+    let df = df.select_columns(&["a"])?;
+
+    // print the results
+    df.show().await?;
```
Comment (comphead) [inline datafusion-examples/examples/dataframe.rs:249]: just a nit: would be `assert` better here so the user can understand the output without running the example?

Comment (alamb) [reply datafusion-examples/examples/dataframe.rs:249]: Good idea -- done in [e06d83f](https://github.com/apache/datafusion/pull/13939/commits/e06d83ff4729d111484911238eadbf06a551f7e8)

**Comment by alamb** [comment]
Thanks again @comphead for the review

**Pull request merged**